dspace-api/src/main/java/org/dspace/disseminate/CitationDocumentServiceImpl.java

```diff
@@ -297,8 +296,21 @@ public boolean canGenerateCitationVersion(Context context, Bitstream bitstream)
     }
 
     @Override
-    public Pair<InputStream, Long> makeCitedDocument(Context context, Bitstream bitstream)
-            throws IOException, SQLException, AuthorizeException {
+    public InputStream getCitedDocument(Context context, Bitstream bitstream)
+        throws SQLException, AuthorizeException, IOException {
+        byte [] citedDocumentContent = getCitedDocumentContent(context, bitstream);
+        return new ByteArrayInputStream(citedDocumentContent);
+    }
+
+    @Override
+    public Long getCitedDocumentLength(Context context, Bitstream bitstream)
+        throws SQLException, AuthorizeException, IOException {
+        byte[] citedDocumentContent = getCitedDocumentContent(context, bitstream);
+        return Long.valueOf(citedDocumentContent.length);
```

Comment:
splitting this in two separate methods is inefficient as it means that the coverpage and the resulting file must be generated potentially two times. I recommend to keep the current service interface as is so that a single method provide both information. Eventually we can accept this code as a quick fix for the 7.2 and improve it later to avoid calling the method two times